### ISSUE TITLE
Add resource-file support in config.xml

### ISSUE THREAD

**Issue by yimingkuan** [phonegap/build#472]
`gap:config-file` allows users to edit the manifest/plist file, but a common request for some users for adding custom images/strings/etc or other resource files to their PhoneGap platform projects. Common examples include overriding images added by plugins for UX purposes, string translations for different users on Android, etc.

This might only be implementable after #445 though.

**Comment by mauron85** [comment]
+1

**Comment by mauron85** [comment]
Might be helpful. The biggest impact will be for cordova plugins like phonegap-plugin-push, which allows custom icons in the drawer. However PhoneGap Build users are out of luck, till this is implemented.

http://stackoverflow.com/questions/30802589/how-to-add-native-image-with-phonegap-build
http://community.phonegap.com/nitobi/topics/can-i-add-custom-images-in-the-drawable-directory
https://github.com/phonegap/phonegap-plugin-push/issues/118

Thank you.

**Comment by kentmw** [comment]
+1 this is very much needed

**Comment by benmcmaster** [comment]
+1 !!!!

**Comment by jskrepnek** [comment]
+1 - would like this so that universal links (via https://github.com/nordnet/cordova-universal-links-plugin) can be used.

**Comment by shamank** [comment]
+1 - I'm still stuck with the nasty workaround of using my app image with transparencies just because I can't add a custom image to resources directory, which phonegap-plugin-push requires for the new android layout: https://github.com/phonegap/phonegap-plugin-push/issues/118

**Comment by SolveSoul** [comment]
+1

**Comment by wildabeast** [comment]
A beta feature was added in November that should actually solve this for Android -- its described for the purpose of translation files but should work for arbitrary images etc. See [this issue comment](https://github.com/phonegap/build/issues/401#issuecomment-156277041).

**Comment by shamank** [comment]
@wildabeast thanks a lot! I keep an eye on http://phonegap.com/blog/phonegap-build/ but I'm pretty sure this was not announced. Am I wrong? This is a really key new feature!

**Comment by shamank** [comment]
it works! thanks!

**Comment by neffnet** [comment]
I'm here from the above referenced issue, trying to make a custom sound play when my app receives a push notification in the background. We build with PhoneGap Build and don't know how to place the sound files so that the OS can play them when the app is in background state. @wildabeast comment helped get our custom sound working for android devices, but as far as I know there is no solution to this problem yet for iOS.

**Comment by wildabeast** [comment]
@neffnet yes unfortunately no support for iOS at this time. I'd suggest creating a new issue, and perhaps reference all the pertinent issues in other projects.

**Comment by razam217** [comment]
We are trying to include a sound file in ios project, we have copied it using gulp task but it seems that ios project is not adding this file in its solution, unless we manually add it through xcode. We have tried <resource-file> in config.xml but in vain. If someone has any solution please share.

**Comment by razam217** [comment]
Hey Guys, I've found solution to my problem. I'm adding my custom sound file in **www** directory, and after that when I build my project, it is easily available in XCode **www** directory. 
The only thing you need to do is to add a **www/sound.caf** path in you Push Notification **push**.

**Comment by gabaum10** [comment]
@razam217  FYI, this is not for the Phonegap CLI, but rather the Adobe Phonegap Build service.  Your solution isn't applicable here because Phonegap build users don't need to use Xcode at all.

**Comment by razam217** [comment]
@gabaum10 May be my problem is other than yours, thats why I mentioned **I've found solution to my problem**...
So if anyone facing problem I was facing, can fix it with my solution... 
Thanks

**Comment by jantonso** [comment]
Hi,

I'm trying to add png files like described above from:
<www.zip>/locales/android/image.png   --> <android_apk>/res/image.png

But they are not being copied over when I look in the apk. I'm using phonegap build cli 5.4.1.

Any ideas as to why they aren't getting copied?

Thanks

**Comment by lunaru** [comment]
@razam217 For what it's worth, your answer helped me -- for some simple things the www folder is added to the project by convention so you don't need to specifically config anything to be copied over.

**Comment by Heshyo** [comment]
Is the copy `<www.zip>/locales/android/somedir/myimage.png --> <android_apk>/res/somedir/myimage.png` still implemented on PGB? Using CLI 7.1.0 I can't seem to get my file copied.

I have `www/locales/android/drawable/pushnotificationicon.png` and I zip the content under `www` but I don't see my png in the generated apk.

**Comment by Heshyo** [comment]
It looks like it used to work until CLI 7.0.1 with builder 1 but not 7.0.1 builder 2 and upward: [Copy files into res/values/](https://forums.adobe.com/thread/2417785)

Has this feature been removed or modified?

**Comment by jskrepnek** [comment]
According to the blog, support for `resource-file` was added with version 7.0.1 and the new builder.  @wildabeast We are using `7.1.0` with the **old** builder.  Should the existing _locales_ approach still work?

**Comment by andersborgabiro** [comment]
Why don't you use 7.1.0 or 8.0.0? 7.0.1 is terribly old (a year) :).

**Comment by jskrepnek** [comment]
@andersborgabiro Sorry, that was a typo (I've updated).  We're using version 7.1.0 with the *old* builder.

Have you had any luck with the `resource-file` element?  Or the previous `locales` copy feature?

**Comment by andersborgabiro** [comment]
No. I use my own platform independent translation solution, that stores texts in a JS object.

**Comment by AlexPashley** [comment]
I've just tried to add resource-file element in config.xml with my Phonegap build v8.0.0 and build fails on android

    <resource-file src="push.png" target="res/push.png" />

**Build Log**
Failed to install 'cordova-plugin-device': Error: ENOENT: no such file or directory, open '/AndroidManifest.xml'

**See this for explanation**
https://github.com/whiteoctober/cordova-plugin-app-version/issues/99#issuecomment-359401073

Seen as I cant use hooks any ideas?

**Comment by postersw** [comment]
If you are using PhoneGap build v 8.0.0, your resource-file directive should copy to 'app/src/main/res/:

`<resource-file src="push.png" target="app/src/main/res/push.png" />` 

Copying to res/ will cause PhoneGap build to fail, as noted above.
